cdap-ui/app/cdap/components/PipelineContextMenu/index.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React from 'react';
+import { ContextMenu, IContextMenuOption } from 'components/ContextMenu';
+import WranglerConnection from 'components/PipelineContextMenu/WranglerConnection';
+import If from 'components/If';
+import PropTypes from 'prop-types';
+import { getValueFromClipboard } from 'services/Clipboard';
+import { objectQuery } from 'services/helpers';
+import { INode, IConnection } from 'components/PipelineContextMenu/PipelineTypes';
+import { INewWranglerConnection } from 'components/PipelineContextMenu/WranglerConnection';
+import { GLOBALS } from 'services/global-constants';
+
+export interface IStage {
+  nodes: INode[];
+  connections: IConnection[];
+}
+interface IPipelineContextMenuProps {
+  onNodesPaste: (stages: IStage) => void;
+  onWranglerSourceAdd: INewWranglerConnection;
+  pipelineArtifactType: 'cdap-data-pipeline' | 'cdap-data-streams';
```

Comment:
use of GLOBALS here

Reply:
Its not possible in this case as these are typescript type definition and for some reason ts won't access a object deference in type definition